_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp

```diff
@@ -2962,6 +2962,25 @@ mfxStatus VideoVPPHW::Close()
 } // mfxStatus VideoVPPHW::Close()
 
 
+mfxStatus VideoVPPHW::IsCopyPassThroughPreferred(bool &preferred) const
+{
+    if (m_pCore->GetVAType() != MFX_HW_VAAPI || m_ioMode != D3D_TO_D3D)
```

Comment:
Doesn't this check mean that vaapi core is used?
I.e. cast on line 2979 wouldn't fail?

Reply:
@mgonchar  Yes, the cast must not fail. I don't like dropping the check, however. What can I do here, as there is no asserts in release and you think returning mfxStatus is ugly?

Reply:
well, currently I see two different ways:
1) Leave the current code, but return false instead of error if cast fails (which is supposed to be almost impossible situation). So we just silently hide the error (not actually so, because we can use trace macro to at least signal to stdout if traces are on)
2) Move to references casting, then some upper layers should deal with possible exception (which is supposed to be almost impossible situation). So we loose information about place of error occurrence. 

So both of these ways are definitely not much better than current approach. I will try to figure out better solution...

Reply:
Asserts _are_ enabled in release builds of master branch. They are switched off in builds of release branches. So, use assert if you feel it right.

Reply:
Here I want to be protected from some fearless code refactoring. So checking on master' release is just fine.

Reply:
Having virtual `GetCmCopyStatus` in core is not an option to avoid this cast which is ugly on its own?

Reply:
I believe adding new virtual functions to `CommonCORE` is prohibited. That's why the cast was applied here.